test/unit/http/preprocessors.js

```diff
@@ -272,6 +272,30 @@ describe('preprocessors', () => {
           context.auth.session.should.eql(Option.of(new Session({ test: 'session' })));
         }));
 
+      it('should reject cookie which cannot be url-decoded', () => {
+        let caught = false;
+        Promise.resolve(authHandler(
```

Comment:
I think this promise needs to be returned in order for Mocha to wait on it.

Reply:
Very possible.  Do tests on line 188 and 208 need the same change?

Reply:
Ah good catch, that sounds right to me. If there's an assertion in a promise chain, that assertion will only affect the success/failure of the test if the promise is returned to Mocha.

Reply:
Existing tests fixed at https://github.com/getodk/central-backend/pull/1286